File: UI/skillpath_shared/lib/models/auth/register_request.dart
class RegisterRequest {
  final String firstName;
  final String lastName;
  final String email;
  final String password;
  final String? phoneNumber;
  final int? cityId;

  RegisterRequest({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.password,
    this.phoneNumber,
    this.cityId,
  });

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'firstName': firstName,
      'lastName': lastName,
      'email': email,
      'password': password,
    };
    if (phoneNumber != null) json['phoneNumber'] = phoneNumber;
    if (cityId != null) json['cityId'] = cityId;
    return json;
  }
}

File: SkillPath.Services/DTOs/Auth/RegisterRequest.cs
using System.ComponentModel.DataAnnotations;

namespace SkillPath.Services.DTOs.Auth
{
    public class RegisterRequest
    {
        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string LastName { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        [MinLength(4)]
        public string Password { get; set; } = string.Empty;

        public string? PhoneNumber { get; set; }

        public int? CityId { get; set; }
    }
}
